aiutils/tftools/placeholder_management.py

```diff
@@ -61,12 +70,25 @@ def add_placeholder(self, name, dtype, shape=None, sparse=False):
             name_values = name + '_values'
             name_shape = name + '_shape'
 
-            self._placeholders[name_indices] = tf.placeholder(tf.int64, [None, 2],
-                                                           name_indices)
+            self._placeholders[name_indices] = tf.placeholder(
+                tf.int64, [None, 2], name_indices)
             self._placeholders[name_values] = tf.placeholder(dtype, [None],
-                                                          name_values)
+                                                             name_values)
             self._placeholders[name_shape] = tf.placeholder(tf.int64, [2],
-                                                         name_shape)
+                                                            name_shape)
+
+    def add_list_placeholder(self, name, dtype, size, shape=None,
+                             sparse=False):
+        assert_string = 'size needs to be a positive integer'
+        assert (isinstance(size, int) and size > 0), assert_string
+        self.islist[name] = size
+        for i in xrange(size):
```

Comment:
Use `from builtins import range` and then range rather than xrange (for python3 compatibility)